### PR TITLE
Add support for ACME revoke

### PR DESCRIPTION
**Pull request by musgravejw**
Related: https://github.com/onmodulus/internal/issues/161

### CODE REVIEW

**Comment by jackboberg** [comment]
👍 LGTM

File: lib/ACMEClient.js [hunk 1]
```diff
@@ -7,23 +7,34 @@ const base64url = require("./base64url");
...
     }
-    
+
+    revoke(cert) {
+      return this.requester.request("/acme/revoke-cert", {
+        resource: "revoke-cert",
+        certificate: convertDERtoPEM(cert),
+      }).then((result) => {
+          if (result.statusCode != 201 && result.statusCode != 409) {
+              throw new Error("Error registering: " + result.statusCode + " " + result.body);
```
Comment (jackboberg) [inline lib/ACMEClient.js]: probably should be `Error revoking`

**Comment by franvarney** [comment]
What is the status of this?

**Comment by musgravejw** [comment]
Untested, and doesn't benefit our Let's Encrypt client.

**Comment by theworkflow** [comment]
@musgravejw if this is no longer a relevant PR can you close it?

**Pull request closed without merging**